feat(docs): introduced navigation system based on #169 (#2606)
nodejs/nodejs.dev: ADDING_LEARN_PAGE.md

## When you need to follow this process

With the working group iterating on design and content structure, this process is our way of making sure all content gets good feedback and, with approval, eventually finds its way to the right place in the documentation.

You need to follow this process if you intend to make a "substantial" addition or change to the proposed Node.js getting started guides in this repo. What constitutes a "substantial" change evolves based on community norms. Some changes do not require an RFC. These may include reasonably sized re-phasing, reorganizing, or refactoring an existing documentation page.

If you submit a pull request to implement a significant change to the documentation without going through the RFC process, it may be closed with a polite request to submit an RFC first!

## To begin drafting a documentation page:

> 1. Fork the [nodejs/website-redesign repo](https://github.com/nodejs/website-redesign) so that you have your own copy of the repository that you have push access to.

Author: ```suggestion
1. Fork the [nodejs/nodejs.dev repo](https://github.com/nodejs/nodejs.dev) so that you have your own copy of the repository to which you have push access.
```

> 1. Fork the [nodejs/website-redesign repo](https://github.com/nodejs/website-redesign) so that you have your own copy of the repository that you have push access to.

Author: Nope, updated. Good catch.

> 1. Create your documentation file on `content/learn` use a reasonable and descriptive name for the file.

Author: You don't need to add `2.` here. As per Markdown docs, adding `1.` everywhere enables automatic paging.

> 1. Create your documentation file on `content/learn` use a reasonable and descriptive name for the file.

Author: ```suggestion
1. Create your documentation file in `content/learn`. Use a reasonable and descriptive name for the file.
```

  - If you're writing a simple Markdown file, use an `.md` extension. Otherwise, if you want to incorporate [JSX](https://reactjs.org/docs/introducing-jsx.html) features, sabve it with an `.mdx` extension.

> 1. Copy the metadata from the template below. This metadata defines crucial information such as authors, the title of your documentation page, and more

Author: ```suggestion
1. Copy the metadata from the template below. This metadata defines crucial information such as authors, the title of your documentation page, and more.
```

  ```mdx
  ---
  title: 'A title for your page'
  description: 'A short description of your page'
  authors: your-github-username
  section: Name of the Section # Available categories are 'Quick Start' and 'Getting Started'
  category: learn # Leave this as it is
  ---
  ```

> 1. Edit your file and begin writing out your article. Put care into the details! Please review the proposed [Node.js Voice and Tone Guidelines](https://github.com/nodejs/website-redesign/blob/master/style-guide/0001-voice-and-tone.md) before you begin writing.

Author: ```suggestion
1. Edit your file and begin writing out your article. Put care into the details! Please review the proposed [Node.js Voice and Tone Guidelines](https://github.com/nodejs/nodejs.dev/blob/main/style-guide/0001-voice-and-tone.md) before you begin writing.
```

1. When you are ready to get some feedback on your article, submit a pull request back to the main nodejs/nodejs.dev repository. As a pull request, the draft will receive feedback from the larger community, and the author should be prepared to revise it in response.
1. Build consensus on your article by integrating feedback from the community.
1. Eventually, the Website Redesign Working Group will decide in one of their bi-weekly meetings whether the RFC is a candidate for inclusion on the Node.js website.

### Recommendations

> - We recommend you to give a read to general Markdown and MDX documentation before you start writing your article.

Author: ```suggestion
- We recommend you read general Markdown and MDX documentation before you start writing your article.
```
